tests/test_emulation.py

```diff
@@ -0,0 +1,211 @@
+        await page.setViewport({'width': 100, 'height': 100})
+        assert await page.evaluate('"ontouchstart" in window') is False
+
+    @chrome_only
+    @sync
+    async def test_detects_touch(self, server, isolated_page):
+        """It should detect touch when applying viewport with touches."""
+        page = isolated_page
+        await page.goto(server / 'detect-touch.html')
+        assert await page.evaluate('document.body.textContent.trim()') == 'NO'
+        await page.setViewport(iPhone['viewport'])
+        assert await page.evaluate('document.body.textContent.trim()') == 'YES'
+
+    @chrome_only
+    @sync
+    async def test_is_detecteable_by_modernizr(self, server, isolated_page):
+        """Verify the emulation should be detectable by Modernizr JS lib."""
+        page = isolated_page
+        await page.setViewport({'width': 800, 'height': 600, 'hasTouch': True})
+        await page.addScriptTag(url=server / 'modernizr.js')
+        assert await page.evaluate('Modernizr.touchevents') is True
+
+    @chrome_only
+    @sync
+    async def test_supports_landscape_emulation(self, server, isolated_page):
+        """It should support landscape emulation."""
+        page = isolated_page
+        await page.goto(server / 'mobile.html')
+        assert await page.evaluate('screen.orientation.type') == 'portrait-primary'
+        await page.setViewport(iPhoneLandscape['viewport'])
+        assert await page.evaluate('screen.orientation.type') == 'landscape-primary'
+        await page.setViewport({'width': 100, 'height': 100})
+        assert await page.evaluate('screen.orientation.type') == 'portrait-primary'
+
+
+class TestPageEmulation:
+    @sync
+    async def test_page_emulates_mobile_devices(self, server, isolated_page):
+        """Verify the page emulation works with emulating mobile devices."""
+        page = isolated_page
+        await page.goto(server / 'mobile.html')
+        await page.emulate(viewport=iPhone['viewport'], userAgent=iPhone['userAgent'])
+        assert await page.evaluate('window.innerWidth') == 375
+        assert 'iPhone' in await page.evaluate('navigator.userAgent')
+
+    @sync
+    async def test_supports_clicking(self, server, isolated_page):
+        """Verify the emulation should support clicking."""
+        page = isolated_page
+        await page.emulate(viewport=iPhone['viewport'], userAgent=iPhone['userAgent'])
+        await page.goto(server / 'input/button.html')
+        button = await page.J('button')
+        await page.evaluate('button => button.style.marginTop = "200px"', button)
+        await button.click()
+        assert await page.evaluate('result') == 'Clicked'
+
+
+class TestEmulateMediaType:
+
+    @chrome_only
+    @sync
+    async def test_emulation_media_type(self, isolated_page):
+        """The emulation media type should work."""
+        page = isolated_page
+        assert await page.evaluate('matchMedia("screen").matches') is True
```

Comment:
We can just leaves these truthy values be:
```suggestion
        assert await page.evaluate('matchMedia("screen").matches')
```
Same goes for the others.